index.js

```diff
@@ -1,7 +1,43 @@
 const visit = require('unist-util-visit');
 var u = require('unist-builder');
 
+const TAG_SPLIT = /([^<]+)?(<[^>]+>)*/
+const mightHaveTag = str => str ? str.indexOf('<') < str.indexOf('>') : false;
+
+let recursion = 0;
+const splitTags = (str) => {
+  if (recursion > 4) return [str];
+  if (!mightHaveTag(str)) return [str];
+
+  let [_, ...result] = str.match(TAG_SPLIT)
+
+  result = (result || []).filter(Boolean).reduce((acc, subStr, i) => {
+    let previous = acc[i - 1];
+    let previousExists = previous !== undefined;
+    let previousIsComponent = previousExists && isComponentInvocationOrHandlebars(previous)
+
+    if (i > 0 && !isComponentInvocationOrHandlebars(subStr) && !previousIsComponent) {
+      acc[i - 1] = `${previous || ''}${subStr}`;
+    } else {
+      acc.push(subStr);
+    }
+
+    return acc;
+  }, []);
+
+  if (result.length === 1) {
+    return [str];
+  }
+
+  if (result.length > 1) {
+    result = result.map(subStr => splitTags(subStr));
```

Comment:
You can simplify this a bit:

```suggestion
    result = result.map(splitTags);
```